src/chainrules.jl

```diff
@@ -1,8 +1,43 @@
-function ChainRulesCore.rrule(::typeof(*), A::AbstractSparseMatrix, B::DenseInputVecOrMat)
+function ChainRulesCore.rrule(
+    ::typeof(*), A::AbstractSparseMatrix{T}, B::DenseInputVecOrMat
+) where T 
     C = A * B
     function mul_pullback(C̄)
         _, i_A, i_B, _, _ = (~imul!)(AD.GVar(C, C̄), AD.GVar(A), AD.GVar(B), AD.GVar(1.), AD.GVar(1.))
         return ChainRulesCore.NoTangent(), AD.grad(i_A), AD.grad(i_B)
     end
     return C, mul_pullback
 end
+
+function ChainRulesCore.rrule(
+    ::typeof(*), xA::Adjoint{T, <:AbstractSparseMatrix}, B::DenseInputVecOrMat
+) where T
+    C = xA * B
+    function mul_pullback(C̄)
+        _, i_xA, i_B, _, _ = (~imul!)(AD.GVar(C, C̄), AD.GVar(xA), AD.GVar(B), AD.GVar(1.), AD.GVar(1.))
+        return ChainRulesCore.NoTangent(), AD.grad(i_xA), AD.grad(i_B)
+    end
+    return C, mul_pullback
+end
```

Comment:
After we finished the complex case, we need to think about adjoint of sparsecsc. Maybe a new sprand_tangent method is needed?

Reply:
May be our implementation should refer to `ProjectTo` in ChainRulesCore. This is [sparsecsc implementation](https://github.com/JuliaDiff/ChainRulesCore.jl/blob/1014715a38a0f5a9c1e84a804fb1b289e7fb760f/src/projection.jl#L440) and [adjoint implementation](https://github.com/JuliaDiff/ChainRulesCore.jl/blob/1014715a38a0f5a9c1e84a804fb1b289e7fb760f/src/projection.jl#L440). I'm not sure if both of the methods were added, would it adapt to `Adjoint{SparseCSC}`? Or we should implement `sprand_tangent` similar to `ProjectTo` on `Adjoint{SparseCSC}` type?